azure-toolkit-libs/azure-toolkit-containerapps-lib/src/main/java/com/microsoft/azure/toolkit/lib/containerapps/task/DeployContainerAppTask.java

```diff
@@ -0,0 +1,116 @@
+package com.microsoft.azure.toolkit.lib.containerapps.task;
+
+import com.microsoft.azure.toolkit.lib.Azure;
+import com.microsoft.azure.toolkit.lib.common.bundle.AzureString;
+import com.microsoft.azure.toolkit.lib.common.exception.AzureToolkitRuntimeException;
+import com.microsoft.azure.toolkit.lib.common.model.Region;
+import com.microsoft.azure.toolkit.lib.common.operation.AzureOperation;
+import com.microsoft.azure.toolkit.lib.common.task.AzureTask;
+import com.microsoft.azure.toolkit.lib.containerapps.AzureContainerApps;
+import com.microsoft.azure.toolkit.lib.containerapps.config.ContainerAppConfig;
+import com.microsoft.azure.toolkit.lib.containerapps.config.ContainerAppsEnvironmentConfig;
+import com.microsoft.azure.toolkit.lib.containerapps.containerapp.ContainerApp;
+import com.microsoft.azure.toolkit.lib.containerapps.containerapp.ContainerAppDraft;
+import com.microsoft.azure.toolkit.lib.containerapps.environment.ContainerAppsEnvironment;
+import com.microsoft.azure.toolkit.lib.containerapps.environment.ContainerAppsEnvironmentDraft;
+import com.microsoft.azure.toolkit.lib.resource.AzureResources;
+import com.microsoft.azure.toolkit.lib.resource.ResourceGroup;
+import com.microsoft.azure.toolkit.lib.resource.task.CreateResourceGroupTask;
+import org.apache.commons.lang3.StringUtils;
+
+import javax.annotation.Nonnull;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+
+public class DeployContainerAppTask extends AzureTask<ContainerApp> {
+    private static final String EXPANDABLE_REGION_WARNING = "'%s' may not be a valid region, " +
+        "please refer to https://aka.ms/maven_webapp_runtime#region for valid values";
+
+    private final ContainerAppConfig config;
+    private ContainerApp containerApp;
+    @Nonnull
+    private final List<AzureTask<?>> subTasks;
+    public DeployContainerAppTask(ContainerAppConfig config) {
+        this.config = config;
+        this.subTasks = new ArrayList<>();
+        this.initTasks();
+    }
+    private void initTasks() {
+        final ContainerAppsEnvironmentConfig environmentConfig = config.getEnvironment();
+        final String subscriptionId = Optional.ofNullable(environmentConfig).map(ContainerAppsEnvironmentConfig::getSubscriptionId).filter(StringUtils::isNotBlank).orElseThrow(() -> new AzureToolkitRuntimeException("'subscriptionId' is required"));
```

Comment:
The logic here is merely the same: to get the value out of a config, and throw if null. Better to extract to a method.